.github/workflows/continuous-integration-workflow.yml

```diff
@@ -0,0 +1,310 @@
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-yarn
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Run Operator UI tests
+        run: ./tools/ci/operator_ui_test
+
+  explorer:
+    name: Explorer
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+          # Set health checks to wait until postgres has started
+          options: >-
+            --health-cmd pg_isready
+            --health-interval 10s
+            --health-timeout 5s
+            --health-retries 5
+    env:
+      TYPEORM_NAME: ci
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-yarn
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Run Yarn migrations
+        run: yarn workspace @chainlink/explorer test:migration:run
+      - name: Run Server Tests
+        run: yarn workspace @chainlink/explorer test:ci
+      - name: Run Client dependency setup
+        run: yarn workspace @chainlink/explorer-client setup
+      - name: Run Client tests
+        run: yarn workspace @chainlink/explorer-client test:ci
+
+  feeds:
+    name: Feeds
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-yarn
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Run Feeds setup
+        run: yarn workspace @chainlink/feeds setup
+      - name: Run Feeds tests
+        run: yarn workspace @chainlink/feeds test
+        env:
+          CI: true
+
+  reportcoverage:
+    name: Report test coverage
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    needs: [go-postgres, solidity, operator-ui, rust, explorer]
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Init GCloud
+        run: ./tools/ci/init_gcloud
+      - name: Run coverage report
+        run: ./tools/ci/report_coverage
+
+  check_if_rebase_needed:
+    name: Check if rebase needed
+    runs-on: ubuntu-latest
+    container: alpine/git
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+        with:
+          # Fetch entire history. Slower than default, but necessary to check
```

Comment:
Do we really need entire history? Part of gitflow is to have short lived branches, I can't imagine a branch having more than 1k commits, and that would still be very conservative.